### PR TITLE
Use morph tag library for f:file

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/jenkins/pull/7452#discussion_r1060738108

I've updated the docs in https://github.com/jenkinsci/design-library-plugin/pull/193 for this

### Testing done

Tested with `f:file` in `PluginManager/advanced.jelly` and `FileParameterDefinition/index.jelly`

Added `clazz` attribute and checked class was there as expected

### Proposed changelog entries

- Developer: `f:file` now uses the morph tag library, all unknown attributes will be copied to the element.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7562"></a>

### CODE REVIEW

File: core/src/main/resources/lib/form/file.jelly
Comment (timja) [inline core/src/main/resources/lib/form/file.jelly:36]: Annoyingly the IntelliJ plugin doesn't support the morph tag library and marks anything using accept as an error 😢.

image

File: core/src/main/resources/lib/form/file.jelly
Comment (NotMyFault) [inline core/src/main/resources/lib/form/file.jelly:41]:
```suggestion
    <st:attribute name="name">
      This becomes @name of the <input> tag.
      If @field is specified, this value is inferred from it.
```
There's no data we need to escape.

**Review by NotMyFault — APPROVED** [review]

File: core/src/main/resources/lib/form/file.jelly
Comment (timja) [reply core/src/main/resources/lib/form/file.jelly:41]: sure was just copy pasted from textbox.jelly I think

**Review by strangelookingnerd — APPROVED** [review]

**Comment by NotMyFault** [comment]
No idea what I took a look at before, angle brackets need to be escaped, hence https://github.com/jenkinsci/jenkins/pull/7562/commits/f1c661da8f47ea288c760a319ce096fa94f660d7 🤦🏻 myself

**Review by jglick — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by strangelookingnerd — APPROVED** [review]

File: core/src/main/resources/lib/form/file.jelly
Comment (daniel-beck) [inline core/src/main/resources/lib/form/file.jelly:37]: Even though no longer explicit in the code, this is the only element this applies to per https://github.com/jenkinsci/jenkins/blob/c3e29a4398adf625600746f99989d18570608fc6/war/src/main/webapp/scripts/hudson-behavior.js#L2422 and so this change removes the only documentation there is for it.

(If instead support is dropped, it should be removed from callers and `hudson-behavior.js` as well.)

**Review by daniel-beck — DISMISSED** [review]

**Pull request merged**
